File: calc_je.py
import numpy as np
from calc_jte import calc_jte
from calc_jre import calc_jre
from Get_global_value import num_q


def calc_je(RR, AA, q, joints):
    n = len(joints)
    JJ = np.zeros((n, 6))
    JJ_te = calc_jte(RR, AA, q, joints)
    JJ_re = calc_jre(AA, joints)
    for i in range(n):
        JJ[i, 0:3] = JJ_te[i, 0:3]
        JJ[i, 3:6] = JJ_re[i, 0:3]

    Jacobian = np.zeros((num_q, 6))
    for i in range(n):
        Jacobian[joints[i], :] = JJ[i, 0:6]

    return Jacobian

File: src/matlab/spacedyn_v2r1/calc_je.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% The SpaceDyn, a MATLAB toolbox for Space and Mobile Robots.
% (C)1998 The Space Robotics Lab. directed by Kazuya Yoshida,
% Tohoku University, Japan.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%	February 4, 1998, Last modification by K.Yoshida
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
%
% CALC_JE	Calculation of the Jacobian Matrix
%           for the endpoint given by connection vector 'joints.'
%
%   Space Robotics Lab (C)
%   1997.1.14 A.Kurosu
%   2002.9.11 H.Hamano
%   2002.2.27 H.Hamano modified to version up
%

function Jacobian = calc_je( LP, SV, joints )

global Ez


% number of links from base to endpoint.
n = length(joints);

% Calculation of Jacobian
JJ_te = zeros(3,LP.num_q);
JJ_re = zeros(3,LP.num_q);

JJ_te = calc_jte( LP, SV, joints );
JJ_re = calc_jre( LP, SV, joints );
JJ = [ JJ_te; JJ_re ];

% Compose the Jacobian using the corresponding joints.
Jacobian = zeros(6,LP.num_q);

for i = 1 : 1 : n
   Jacobian(1:6 , joints(i)) = JJ(1:6 , i);
end


%%%EOF
